### ISSUE TITLE
Bower Search Website is Broken

### ISSUE THREAD

**Issue by kumarharsh** [bower/bower#1778]
As of today, the bower search website is completely broken.
This is the error I get on opening the page:

```javascript
Error: group[0].website is undefined
[7]</module.exports/</dedupe/<@http://bower.io/search/dist/app.js:12:10067
runInContext/forOwn@http://bower.io/search/dist/app.js:11:28170
forEach@http://bower.io/search/dist/app.js:11:2959
[7]</module.exports/</dedupe@http://bower.io/search/dist/app.js:12:9987
[7]</module.exports/</$scope.search@http://bower.io/search/dist/app.js:12:10857
[7]</module.exports/<@http://bower.io/search/dist/app.js:12:11728
invoke@http://bower.io/search/dist/app.js:2:3636
instantiate@http://bower.io/search/dist/app.js:2:3861
$ControllerProvider/this.$get</<@http://bower.io/search/dist/app.js:3:8735
nodeLinkFn/<@http://bower.io/search/dist/app.js:2:24067
forEach@http://bower.io/search/dist/app.js:1:9010
nodeLinkFn@http://bower.io/search/dist/app.js:2:23738
compositeLinkFn@http://bower.io/search/dist/app.js:2:15683
compile/<@http://bower.io/search/dist/app.js:2:14601
bootstrap/doBootstrap/</<@http://bower.io/search/dist/app.js:1:18313
$RootScopeProvider/this.$get</Scope.prototype.$eval@http://bower.io/search/dist/app.js:5:19223
$RootScopeProvider/this.$get</Scope.prototype.$apply@http://bower.io/search/dist/app.js:5:19576
bootstrap/doBootstrap/<@http://bower.io/search/dist/app.js:1:18254
invoke@http://bower.io/search/dist/app.js:2:3636
bootstrap/doBootstrap@http://bower.io/search/dist/app.js:1:18134
bootstrap@http://bower.io/search/dist/app.js:1:18447
angularInit@http://bower.io/search/dist/app.js:1:17625
[2]</</<@http://bower.io/search/dist/app.js:10:13245
trigger@http://bower.io/search/dist/app.js:7:20362
createEventHandler/eventHandler/<@http://bower.io/search/dist/app.js:1:31507
forEach@http://bower.io/search/dist/app.js:1:8856
createEventHandler/eventHandler@http://bower.io/search/dist/app.js:1:31468
```

Also, in the search page, it should be **Your search didn't return any results.** instead of *You search...*

**Comment by sheerun** [comment]
It seems working now?

**Comment by kumarharsh** [comment]
Nopes. Still broken. Angular is breaking, because one of the expressions is throwing.

**Comment by kumarharsh** [comment]
FYI, running **Firefox 39.0a2** on Ubuntu 14.04.

**Comment by sheerun** [comment]
What link are you using. Website is working for me.

**Comment by AStoker** [comment]
I'm having the same issue. Reproducible on Safari for Mac; Chrome, Firefox, and IE on Windows (all current versions). Link being used is http://bower.io/search/ 

Error (when searching for jquery: http://bower.io/search/?q=jquery): 

```javascript
[Error] Error: undefined is not a function (evaluating 'keyword.toLowerCase()')
http://bower.io/search/dist/app.js:12:16897
some@[native code]
http://bower.io/search/dist/app.js:12:16848
every@[native code]
matchesByKeyword@http://bower.io/search/dist/app.js:12:16803
http://bower.io/search/dist/app.js:12:17475
filter@http://bower.io/search/dist/app.js:11:1829
find@http://bower.io/search/dist/app.js:12:17184
search@http://bower.io/search/dist/app.js:12:18834
http://bower.io/search/dist/app.js:12:20000
wrappedCallback@http://bower.io/search/dist/app.js:5:25282
wrappedCallback@http://bower.io/search/dist/app.js:5:25282
http://bower.io/search/dist/app.js:5:18277
$eval@http://bower.io/search/dist/app.js:5:44270
$digest@http://bower.io/search/dist/app.js:5:42128
$apply@http://bower.io/search/dist/app.js:5:44701
done@http://bower.io/search/dist/app.js:3:34918
completeRequest@http://bower.io/search/dist/app.js:3:53927
onreadystatechange@http://bower.io/search/dist/app.js:3:54940
	(anonymous function) (app.js, line 3)
	(anonymous function) (app.js, line 3)
	wrappedCallback (app.js, line 5)
	wrappedCallback (app.js, line 5)
	(anonymous function) (app.js, line 5)
	$eval (app.js, line 5)
	$digest (app.js, line 5)
	$apply (app.js, line 5)
	done (app.js, line 3)
	completeRequest (app.js, line 3)
	onreadystatechange (app.js, line 3)
```

**Comment by kumarharsh** [comment]
I'm using the same link as @AStoker mentioned. The regular [bower.io/search](http://bower.io/search/)

**Comment by AStoker** [comment]
It's worth mentioning that this is no longer happening for me. Tested on Safari and Chrome.

**Comment by nwinkler** [comment]
Tried it earlier today and it's working fine for me, too.

**Comment by samccone** [comment]
yep this was fixed by @desandro

**Comment by kumarharsh** [comment]
OK, it's working in Chrome and Firefox's stable builds. Firefox's Developer build (alpha) never opens the site, even from a cold start with a clean cache

**Comment by AStoker** [comment]
What kind of errors are you getting in the console (if any)? Since it is an alpha build, it may or may not be the Bower site itself.

**Comment by kumarharsh** [comment]
I've already posted the error I got in the original post above. I do think it could be because of the alpha version, but I'm making angular apps myself, and haven't had any errors yet, so it's sort of a head-scratcher really

**Comment by Recon** [comment]
FYI, I had the same problem as @kumarharsh for a while with FF 38.0 (beta) but not with the 39.0a2 (dev edition). I made some digging and saw that `ls.firstPage` value from the local storage of the 38.0 browser (the one with the problem) was having an unusual value: ![see image](http://i.imgur.com/5HjGa00.png) instead of a nice array of objects like in the 39.0a2 one.

I did a `localStorage.removeItem('ls.firstPage')` from the console of the problematic browser to purge malformed value and it's now working again. Unfortunately I am not able to tell what's the cause of this malformed value.
